### PR TITLE
Add update_many/2

### PR DESCRIPTION
**Pull request by dvic**
As discussed in https://github.com/phoenixframework/phoenix_live_view/pull/2362#issuecomment-1638788854

TODO
- [x] tests
- [x] deprecation warning message
- [ ] test on a real project
- [ ] documentation

### CODE REVIEW

File: lib/phoenix_live_component.ex [hunk 1]
```diff
@@ -515,6 +515,9 @@ defmodule Phoenix.LiveComponent do
   @callback mount(socket :: Socket.t()) ::
               {:ok, Socket.t()} | {:ok, Socket.t(), keyword()}
 
+  @callback update_many(list_of_assigns :: [Socket.assigns()], sockets :: [Socket.t()]) ::
+              {:ok, [Socket.t()]}
```
Comment (josevalim) [inline lib/phoenix_live_component.ex:519]: Perhaps?

```suggestion
              [Socket.t()]
```

Comment (dvic) [reply lib/phoenix_live_component.ex]: Yup

**Comment by dvic** [comment]
Oops, the CI is failing on the warning message. Should I do this differently? A message at runtime? Not at compile time?

**Comment by josevalim** [comment]
Continuing in #2771.

**Pull request closed without merging**